File: easy/3442. Maximum Difference Between Even and Odd Frequency I/3442. Maximum Difference Between Even and Odd Frequency I.ts
function maxDifference(s: string): number {
    const cnt: { [key in string]: number } = {}
    for (const c of s) {
        cnt[c] = !cnt[c] ? 1 : cnt[c] + 1
    }
    console.log(cnt)

    let maxOdd = 0
    let minEven = 10 ** 10

    for (const v of Object.values(cnt)) {
        if (v % 2 !== 0) {
            maxOdd = Math.max(maxOdd, v)
        } else {
            minEven = Math.min(minEven, v)
        }
    }

    return maxOdd - minEven
};

File: easy/3442. Maximum Difference Between Even and Odd Frequency I/3442. Maximum Difference Between Even and Odd Frequency I.py
from collections import Counter


class Solution:
    def maxDifference(self, s: str) -> int:
        cnt = Counter(list(s))

        max_odd = 0
        min_even = 10**10
        for _, v in cnt.items():
            if v % 2 != 0 and v > max_odd:
                max_odd = v

            if v % 2 == 0 and v < min_even:
                min_even = v

        return max_odd - min_even


solution = Solution()
print(solution.maxDifference(s="aaaaabbc"))
print(solution.maxDifference(s="abcabcab"))
